### DISCUSSION TITLE
Allow to specify git-dir and working dir from command line

### DISCUSSION THREAD

**Discussion by wl2776** [sourcegit-scm/sourcegit#771] [Ideas]
Example, `sourcegit --git-dir=$HOME/.cfg --work-tree=$HOME`

This is useful for browsing repositories with linux/macOS config files.

Details and recipe: https://www.ackama.com/articles/the-best-way-to-store-your-dotfiles-a-bare-git-repository-explained/

lazygit provides this option

**Comment by wl2776** [comment]
@love-linger, @aikawayataro, could you explain your negative reactions?  Why don't you like this idea?

**Reply by aikawayataro** [reply]
I think this feature does not work well with the existing project-based interface. I also don't see this exact use case as viable.
If you want to separate GIT_DIR you can use `.git` file (not directory) to specify path to GIT_DIR: `echo "gitdir: /path/to/repo.git" > .git`

**Reply by wl2776** [reply]
Indeed, this is an advanced feature, and the one who is performing it should know exactly what is happening.
